### DISCUSSION TITLE
Open modal programmatically - ReferenceError: bootstrap is not defined

### DISCUSSION THREAD

**Discussion by realrecordzLab** [twbs/bootstrap#31239] [General] [answered]
I'm using bootstrap v5 in my vue app. I'm importing the css and it works fine. 
I want to use the modal component and open it programmatically. I'm importing in the `main.js` file of my vue app the `bootstrap.min.js` file using this syntax: `import 'bootstrap/dist/js/bootstrap.min.js'` and in my methods I've created a function that have this code: 
```
        showTableData() {
            let date = new Date().toLocaleDateString()
            let dataExportModal = new bootstrap.Modal(document.getElementById('myModal'))
            dataExportModal.show()
        }
```
What is happening is that I get this error in console when I click on the button that will use the function: `ReferenceError: bootstrap is not defined`

I'm not sure on what's wrong with the code, but any help will be appreciated.

**Comment by Johann-S** [comment]
Hi @realrecordzLab,

There are several way to import Bootstrap. If you're using Vue.js I assume you have a Webpack bundler under the hood.

So you can import our Modal component like that: 

`import { Modal } from 'bootstrap'`

or if  you want to import everything you can do:

`import * as Bootstrap from 'bootstrap'`

**Reply by SanalDominic** [reply]
i am using bootstrap cdn
how can I use it

**Comment by realrecordzLab** [comment]
I've tried to import only the modal component but I get a `bootstrap is not
defined` error. I'm using the modal inside a component, tried to import the
bs javascript also there but same problem, I don't know what's wrong, but
I'm start thinking that it's a bug of the new version of bootstrap?

**Reply by DeoThemes** [reply]
I'm having the same issue. Did you find solution?

**Comment by Johann-S** [comment, marked as answer]
A simple Bootstrap 5 + Vue.js project: https://stackblitz.com/edit/bootstrap5-vuejs

enjoy 😉

**Reply by realrecordzLab** [reply]
I will take a look, I'm on single file components but I will give a new try to the import as you've suggested!

**Reply by AshutoshSharma7** [reply]
Hey Guyz,
I found the solution after a lotof struggle.
Add in application.js
window.bootstrap = require('bootstrap/dist/js/bootstrap.bundle.js')

after this "`bootstrap is not defined" IS 

Ex
import 'bootstrap'
require("@rails/ujs").start()
require("turbolinks").start()
require("@rails/activestorage").start()
require("channels")

window.bootstrap = require('bootstrap/dist/js/bootstrap.bundle.js')

**Comment by AshutoshSharma7** [comment]
Hey Guyz,
I found the solution after a lotof struggle.
Add in application.js
window.bootstrap = require('bootstrap/dist/js/bootstrap.bundle.js')

after this "`bootstrap is not defined" IS 

Ex
import 'bootstrap'
require("@rails/ujs").start()
require("turbolinks").start()
require("@rails/activestorage").start()
require("channels")

window.bootstrap = require('bootstrap/dist/js/bootstrap.bundle.js')

**Comment by deepakmahakale** [comment]
Just replace the following in `app/javascript/packs/application.js`

```js
require("bootstrap")
```

with

```js
window.bootstrap = require("bootstrap")
```

worked for me in rails 6 + webpacker + bootstrap 5
